raiden-ts/src/utils/ethers.ts

```diff
@@ -39,62 +38,73 @@ export function fromEthersEvent<T>(
  * @param contract - Contract source instance for filters, connected to a provider
  * @param filters - array of OR filters from tokenNetwork
  * @param fromBlock$ - Observable of a past blockNumber since when to fetch past events
- * @param lastSeenBlock$ - Observable of latest seen block, to be used as toBlock of pastEvents.
- *      lastSeenBlock + 1 is supposed to be first one fetched by contract.on newEvents$
- *      Both fromBlock$ and lastSeenBlock$ need to be set to fetch pastEvents$
+ *                     If not provided, last resetEventsBlock is automatically used.
  * @returns Observable of contract's events
  */
 export function getEventsStream<T extends any[]>(
   contract: Contract,
-  filters: EventFilter[],
+  filters: Filter[],
   fromBlock$?: Observable<number>,
-  lastSeenBlock$?: Observable<number>,
 ): Observable<T> {
   const provider = contract.provider as JsonRpcProvider;
 
+  const logToEvent = (log: Log): T | undefined => {
+    // parse log into [...args, event: Event] array,
+    // the same that contract.on events/callbacks
+    const parsed = contract.interface.parseLog(log);
+    if (!parsed) return;
+    const args = Array.prototype.slice.call(parsed.values);
```

Comment:
Maybe `parsed.values.slice()`

Reply:
No, this code came from `ethers`, `values` isn't an actual array, but an object with numeric keys and string keys for the parameters and a `length` property, so it is _Array-like_. This `Array.prototype.slice.call` is a way to construct an actual array from this object.